app/forms/sign_in_recaptcha_form.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class SignInRecaptchaForm
+  include ActiveModel::Model
+
+  RECAPTCHA_ACTION = 'sign_in'
+
+  attr_reader :form_class, :form_args, :email, :recaptcha_token, :device_cookie
+
+  validate :validate_device_cookie
+
+  def initialize(form_class: RecaptchaForm, **form_args)
+    @form_class = form_class
+    @form_args = form_args
+  end
+
+  def submit(email:, recaptcha_token:, device_cookie:)
+    @email = email
+    @recaptcha_token = recaptcha_token
+    @device_cookie = device_cookie
+
+    success = valid?
+    FormResponse.new(success:, errors:, serialize_error_details_only: true)
+  end
+
+  private
+
+  def validate_device_cookie
+    return if exempt?
+    recaptcha_response, _assessment_id = recaptcha_form.submit(recaptcha_token)
+    return if recaptcha_response.success?
+    errors.merge!(recaptcha_form)
+  end
+
+  def exempt?
+    device.present?
+  end
+
+  def device
+    User.find_with_confirmed_email(email)&.devices&.find_by(cookie_uuid: device_cookie)
+  end
```

Comment:
This revised lookup is intended to address previous comments from proof-of-concept https://github.com/18F/identity-idp/pull/10587#discussion_r1598443550 and https://github.com/18F/identity-idp/pull/10587#discussion_r1597041921 (cc @mitchellhenke @zachmargolis)